web/src/components/planner/Statistics.vue

```diff
@@ -0,0 +1,327 @@
+              @click="toggleProductsVisibility"
+            >Show all Products
+            </v-btn>
+          </v-col>
+
+          <!-- Produced Items Area -->
+          <div v-show="!hiddenProducts" max-height="500px">
+            <v-divider class="my-4 mx-n4" color="white" thickness="5px" />
+            <h1 class="text-h5 mb-4">
+              <i class="fas fa-conveyor-belt-alt" />
+              <span class="ml-3">Produced Items</span>
+            </h1>
+            <p v-show="helpText" class="mb-4">
+              <i class="fas fa-info-circle" /> Shows all the items produced by all
+              your factories.
+            </p>
+            <div v-if="allFactoryProducts.length > 0">
+              <v-chip
+                v-for="(product) in allFactoryProducts"
+                :key="product.id"
+                class="sf-chip"
+              >
+                <span class="mr-2">
+                  <game-asset :subject="product.id" type="item" />
+                </span>
+                <span>
+                  <b>{{ product.name }}</b>: {{ formatNumber(product.totalAmount) }}/min
+                </span>
+              </v-chip>
+            </div>
+            <p v-else class="text-body-1">Awaiting Production</p>
+          </div>
+        </v-card-text>
+      </v-card>
+    </v-col>
+  </v-row>
+</template>
+
+  <script setup lang="ts">
+  import { computed, ref, watch } from 'vue'
+  import {
+    Factory,
+  } from '@/interfaces/planner/FactoryInterface'
+  import { formatNumber } from '@/utils/numberFormatter'
+  import {
+    getPartDisplayName,
+    hasMetricsForPart,
+  } from '@/utils/helpers'
+
+  const props = defineProps<{
+    factories: Factory[];
+    helpText: boolean;
+  }>()
+
+  // Initialize the 'hidden' ref based on the value in localStorage
+  const hidden = ref<boolean>(
+    localStorage.getItem('statisticsHidden') === 'true'
+  )
+
+  // Watch the 'hidden' ref and update localStorage whenever it changes
+  watch(hidden, newValue => {
+    localStorage.setItem('statisticsHidden', newValue.toString())
+  })
+
+  // Function to toggle visibility
+  const toggleVisibility = () => {
+    hidden.value = !hidden.value
+  }
+  // Initialize the 'hidden' ref based on the value in localStorage
+  const hiddenProducts = ref<boolean>(
+    localStorage.getItem('statisticsProductsHidden') === 'true'
+  )
+
+  // Watch the 'hidden' ref and update localStorage whenever it changes
+  watch(hiddenProducts, newValue => {
+    localStorage.setItem('statisticsProductsHidden', newValue.toString())
+  })
+
+  // Function to toggle visibility
+  const toggleProductsVisibility = () => {
+    hiddenProducts.value = !hiddenProducts.value
+  }
+
+  const getBuildingDisplayName = inject('getBuildingDisplayName') as (
+    part: string
+  ) => string
+
+  // This function calculates total number of buildings for each type of building and creates a new record to store that info in
+  const totalBuildingsByType = computed(() => {
```

Comment:
I'd extract functions like this into their own TS file, e.g. `utils/statistics.ts` simply because testing the vue file logic is a bit hard at the moment.